src/Dhall/TypeCheck.hs

```diff
@@ -504,7 +523,7 @@ typeWithA tpa = loop
                     Const Type -> return ()
                     Const Kind -> return ()
                     _          -> Left (TypeError ctx e (InvalidAlternativeType k t))
-        mapM_ process (Data.HashMap.Strict.InsOrd.toList kts)
+        mapM_ process (Data.HashMap.Strict.toList (Data.HashMap.Strict.InsOrd.toHashMap kts))
```

Comment:
Replacing this with `foldrWithKey` might be even better.

Reply:
#419's benchmark. toList:

```
time                 974.7 ms   (922.6 ms .. 1.012 s)
                     1.000 R²   (0.999 R² .. NaN R²)
mean                 984.7 ms   (977.0 ms .. 989.4 ms)
std dev              7.569 ms   (2.795 ms .. 10.09 ms)
variance introduced by outliers: 19% (moderately inflated)
```

foldrWithKey:

```
time                 952.6 ms   (947.5 ms .. 957.4 ms)
                     1.000 R²   (1.000 R² .. 1.000 R²)
mean                 953.1 ms   (951.8 ms .. 954.6 ms)
std dev              1.791 ms   (672.1 μs .. 2.307 ms)
variance introduced by outliers: 19% (moderately inflated)
```

A 3.3% gain. I guess since I've benchmarked it and found it to be a (small) win, I'll change it.